yb-voyager/src/query/queryissue/issues_ddl.go

```diff
@@ -437,19 +437,438 @@ func NewPrimaryOrUniqueConsOnUnsupportedIndexTypesIssue(objectType string, objec
 	return newQueryIssue(issue, objectType, objectName, sqlStatement, details)
 }
 
-var indexOnComplexDatatypesIssue = issue.Issue{
-	Type:        INDEX_ON_COMPLEX_DATATYPE,
-	Name:        INDEX_ON_COMPLEX_DATATYPE_ISSUE_NAME,
+var indexOnArrayDatatypeIssue = issue.Issue{
+	Type:        INDEX_ON_ARRAY_DATATYPE,
+	Name:        INDEX_ON_ARRAY_DATATYPE_ISSUE_NAME,
 	Impact:      constants.IMPACT_LEVEL_1,
-	Description: INDEX_ON_COMPLEX_DATATYPE_ISSUE_DESCRIPTION,
+	Description: INDEX_ON_ARRAY_DATATYPE_ISSUE_DESCRIPTION,
 	GH:          "https://github.com/yugabyte/yugabyte-db/issues/25003",
-	Suggestion:  INDEX_ON_COMPLEX_DATATYPE_ISSUE_SUGGESTION,
+	Suggestion:  INDEX_ON_ARRAY_DATATYPE_ISSUE_SUGGESTION,
 	DocsLink:    "https://docs.yugabyte.com/preview/yugabyte-voyager/known-issues/postgresql/#indexes-on-some-complex-data-types-are-not-supported",
 }
 
-func NewIndexOnComplexDatatypesIssue(objectType string, objectName string, sqlStatement string, typeName string) QueryIssue {
-	issue := indexOnComplexDatatypesIssue
-	issue.Description = fmt.Sprintf(issue.Description, typeName)
+func NewIndexOnArrayDatatypeIssue(objectType string, objectName string, sqlStatement string) QueryIssue {
```

Comment:
let's add tests for each of these types in issues_ddl_test.go to assert that creating an index actually fails on YB.

Reply:
Added tests